discovery-provider/src/tasks/index_trending.py

```diff
@@ -251,7 +252,14 @@ def get_should_update_trending(
         current_db_block = (
             session.query(Block.blockhash).filter(Block.is_current == True).first()
         )
-        current_block = web3.eth.get_block(current_db_block[0], True)
+        current_block_number = current_db_block[0]
+        final_poa_block = helpers.get_final_poa_block(shared_config)
+        if final_poa_block:
+            nethermind_web3 = web3_provider.get_web3()
+            current_block_number -= final_poa_block
+            current_block = nethermind_web3.eth.get_block(current_block_number, True)
+        current_block = web3.eth.get_block(current_block_number, True)
```

Comment:
should this be in an `else` branch?  Seems this will reassign `current_block` set on line 260